modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java

```diff
@@ -137,7 +139,7 @@ public void handle(MotechEvent event) {
 
         // Handle all tasks one by one
         for (Task task : tasks) {
-            handleTask(task, parameters, false);
+            handleTask(task, parameters, false, retryNumber);
```

Comment:
it would be better to send retryNumber in event SCHEDULE_REPEATING_JOB metadata. Increment this only while handling retry